### PR TITLE
Fix MutualSSLManager should not initialize the keystores in EndpointConfig.properties if the feature is disabled 

### PR DESCRIPTION
**Pull request by hasinidilanka**
### Proposed changes in this pull request

-Fix MutualSSLManager should not initialize the keystores in EndpointConfig.properties if the feature is disabled.

Related to : https://github.com/wso2-support/carbon-identity-framework/pull/467

Issue : https://github.com/wso2/carbon-identity-framework/issues/2039

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

**Comment by hasinidilanka** [comment]
@thariyarox

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/TenantDataManager.java [hunk 1]
```diff
@@ -88,9 +88,11 @@ public static synchronized void init() {
                     inputStream = new FileInputStream(configFile);
 
                     prop.load(inputStream);
-                    if (isSecuredPropertyAvailable(prop)) {
-                        // Resolve encrypted properties with secure vault
-                        resolveSecrets(prop);
+                    if (Boolean.parseBoolean(getPropertyValue(Constants.TenantConstants.TENANT_LIST_ENABLED))) {
```
Comment (thanujalk) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/TenantDataManager.java]: can combine these two if s with a &&

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java [hunk 2]
```diff
@@ -126,42 +130,58 @@ public static synchronized void init() {
...
 
-            if (StringUtils.isNotEmpty(getPropertyValue(Constants.TenantConstants.KEY_MANAGER_TYPE))) {
-                TenantMgtAdminServiceClient.setKeyManagerType(getPropertyValue(Constants.TenantConstants
-                        .KEY_MANAGER_TYPE));
-            }
-            if (StringUtils.isNotEmpty(getPropertyValue(Constants.TenantConstants.TRUST_MANAGER_TYPE))) {
-                TenantMgtAdminServiceClient.setTrustManagerType(getPropertyValue(Constants.TenantConstants
-                        .TRUST_MANAGER_TYPE));
-            }
+                if (StringUtils.isNotEmpty(getPropertyValue(Constants.TenantConstants.TLS_PROTOCOL))) {
```
Comment (thanujalk) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java]: better to check with StringUtils.isNotBlank as these are properties

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java [hunk 3]
```diff
@@ -104,9 +104,13 @@ public static synchronized void init() {
                 try (InputStream inputStream = new FileInputStream(configFile)) {
 
                     prop.load(inputStream);
-                    if (isSecuredPropertyAvailable(prop)) {
-                        // Resolve encrypted properties with secure vault
-                        resolveSecrets(prop);
+                    //Initialize the keystores in EndpointConfig.properties only if the "mutualSSLManagerEnabled"
+                    // feature is enabled.
+                    if (isMutualSSLManagerEnabled(getPropertyValue(Constants.TenantConstants.MUTUAL_SSL_MANAGER_ENABLED))) {
+                        if (isSecuredPropertyAvailable(prop)) {
```
Comment (thanujalk) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java]: Can combine these if s with a &&

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/TenantDataManager.java (hunk 1 above)
Comment (hasinidilanka) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/TenantDataManager.java]: Fixed - https://github.com/wso2/carbon-identity-framework/pull/2559

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java (hunk 2 above)
Comment (hasinidilanka) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java]: Fixed - https://github.com/wso2/carbon-identity-framework/pull/2559

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java (hunk 3 above)
Comment (hasinidilanka) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java]: Fixed - https://github.com/wso2/carbon-identity-framework/pull/2559

**Pull request merged**